### PR TITLE
Check 'modules' directory structure

### PR DESCRIPTION
**Pull request by mirpedrol**
Add a step in all `nf-core modules` commands to check if the modules directory of a pipeline has the correct structure. See slack [discussion](https://nfcore.slack.com/archives/CJRH30T6V/p1664884189665479?thread_ts=1664882848.607819&cid=CJRH30T6V)

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/modules/modules_command.py [hunk 1]
```diff
@@ -147,3 +147,38 @@ def load_lint_config(self):
                 self.lint_config = yaml.safe_load(fh)
         except FileNotFoundError:
             log.debug(f"No lint config file found: {config_fn}")
+
+    def check_modules_structure(self):
+        """
+        Check that the structure of the modules directory in a pipeline is the correct one:
+            'modules/nf-core/TOOL/SUBTOOL
```
Comment (ewels) [inline nf_core/modules/modules_command.py:154]:
```suggestion
            modules/nf-core/TOOL/SUBTOOL

        Prior to nf-core/tools release 2.6 the directory structure had an additional level of nesting:
            modules/nf-core/modules/TOOL/SUBTOOL
```

**Review by ewels — APPROVED** [review]
Reading the code looks good 👍🏻

File: nf_core/modules/modules_command.py [hunk 2]
```diff
@@ -147,3 +147,38 @@ def load_lint_config(self):
...
+                if "main.nf" in files:
+                    module_path = Path(directory).relative_to(Path(self.dir, "modules"))
+                    parts = module_path.parts
+                    # Check that there are modules installed directly under the 'modules' directory
+                    if parts[1] == "modules":
+                        wrong_location_modules.append(module_path)
+            # If there are modules installed in the wrong location
+            if len(wrong_location_modules) > 0:
+                log.info("The modules folder structure is outdated. Reinstalling modules.")
+                # Remove the local copy of the modules repository
```
[the remark this answers is not in the document]
Comment (mirpedrol) [reply nf_core/modules/modules_command.py:168]: I did it like this in case there is a patch file

[the remark this answers is not in the document]
Comment (ewels) [reply nf_core/modules/modules_command.py:168]: This is the cached modules repo though, right? `~/.config/nfcore/nf-core/modules/`?

Patch files are held in the pipeline repo, so the central cache location should be unaffected by those.

[the remark this answers is not in the document]
Comment (mirpedrol) [reply nf_core/modules/modules_command.py:168]: Oh yes you are right, I mixed things. 
So for this I copied code from you 😇 from #1851 
But if there's a better way of doing it, let's change!

**Review by ewels — APPROVED** [review]
Can probably refine the clone update code instead of deleting and redownloading, but can leave that to another day.

**Pull request merged**